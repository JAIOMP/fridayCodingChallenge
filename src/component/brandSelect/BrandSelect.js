import React, {useEffect, useState, useContext} from 'react';
import SelectItem from '../common/selectItem/SelectItem'
import VehicleContext from "../../provider/VehicleDataContext";

const BrandSelect = () => {
    const [brands, setBrands] = useState([]);
    const {setSelectedBrand, setSelectedModel} = useContext(VehicleContext);

    useEffect(() => {
        fetch('http://localhost:8080/api/makes')
            .then(res => res.json())
            .then(data => setBrands(data));
    }, []);

    const onSelect = (event) => {
        setSelectedBrand(event.target.value);
        setSelectedModel("");
    };

    return (
        <>
            <SelectItem data={brands} dataFor="brand" onSelect={onSelect}/>
        </>
    );
};

export default BrandSelect;
