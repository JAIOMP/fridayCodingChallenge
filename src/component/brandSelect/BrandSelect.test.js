import React from 'react';
import {render, waitFor} from '@testing-library/react';
import BrandSelect from './BrandSelect';
import VehicleContext from "../../provider/VehicleDataContext";

describe('BrandSelect', () => {
    const mockResponse = ["ALFA ROMEO", "AUDI", "AUSTIN", "BARKAS", "BMW", "CADILLAC", "CHEVROLET"];

    const contextData = {
        setError: jest.fn(),
        setSelectedModel: jest.fn(),
        selectedBrand: "Ford",
    };
    beforeEach(() => {
        jest.spyOn(global, 'fetch').mockResolvedValue({
            json: jest.fn().mockResolvedValue(mockResponse)
        })
    });

    it('should render BrandSelect', async () => {
        const {container, getByTestId} = render(
            <VehicleContext.Provider value={contextData}>
                <BrandSelect/>
            </VehicleContext.Provider>
        );

        await waitFor(() => getByTestId("select-item"));
        expect(container).toMatchSnapshot();
    });
});
