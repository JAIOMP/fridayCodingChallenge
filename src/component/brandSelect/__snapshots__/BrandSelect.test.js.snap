// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandSelect should render BrandSelect 1`] = `
<div>
  <div
    class="select-item-container"
  >
    <label
      class="select-label"
      data-testid="select-item"
      for="brand"
    >
      Please select your 
      <strong>
        brand
      </strong>
      :
    </label>
    <select
      data-testid="select"
      id="brand-list"
    >
      <option>
        --Please choose an option--
      </option>
      <option
        data-testid="select-option"
        value="ALFA ROMEO"
      >
        ALFA ROMEO
      </option>
      <option
        data-testid="select-option"
        value="AUDI"
      >
        AUDI
      </option>
      <option
        data-testid="select-option"
        value="AUSTIN"
      >
        AUSTIN
      </option>
      <option
        data-testid="select-option"
        value="BARKAS"
      >
        BARKAS
      </option>
      <option
        data-testid="select-option"
        value="BMW"
      >
        BMW
      </option>
      <option
        data-testid="select-option"
        value="CADILLAC"
      >
        CADILLAC
      </option>
      <option
        data-testid="select-option"
        value="CHEVROLET"
      >
        CHEVROLET
      </option>
    </select>
  </div>
</div>
`;
